source/docs/software/basic-programming/reading-stacktraces.rst

```diff
@@ -0,0 +1,553 @@
+ * Access to your source code, (EX: :ref:`on github.com <docs/software/basic-programming/git-getting-started:Git Version Control Introduction>`)
+ * The **full text** of the error, including the full stack trace.
+
+Common Examples & Patterns
+--------------------------
+
+There are a number of common issues which result in runtime exceptions.
+
+Null Pointers and References
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Both C++ and Java have the concept of "null" - a reference which has not yet been initialized, and does not refer to anything meaningful.
+
+Manipulating a "null" reference will produce a runtime error.
+
+For example, consider the following code:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: Java
+          :lineno-start: 19
+
+            PWMSparkMax armMotorCtrl;
+
+            @Override
+            public void robotInit() {
+
+                armMotorCtrl.setInverted(true);
+
+            }
+
+   .. group-tab:: C++
+
+      .. code-block:: C++
+         :lineno-start: 17
+
+         class Robot : public frc::TimedRobot {
+            public:
+               void RobotInit() override {
+                  motorRef->SetInverted(false);
+               }
+
+            private:
+               frc::PWMVictorSPX m_armMotor{0};
+               frc::PWMVictorSPX * motorRef;
+         };
+
+
+When run, you'll see output that looks like this:
+
+.. tabs::
+
+   .. group-tab:: Java
+
+      .. code-block:: text
+
+        ********** Robot program starting **********
+        Error at frc.robot.Robot.robotInit(Robot.java:24): Unhandled exception: java.lang.NullPointerException
+                at frc.robot.Robot.robotInit(Robot.java:24)
+                at edu.wpi.first.wpilibj.TimedRobot.startCompetition(TimedRobot.java:94)
+                at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:335)
+                at edu.wpi.first.wpilibj.RobotBase.lambda$startRobot$0(RobotBase.java:387)
+                at java.base/java.lang.Thread.run(Thread.java:834)
+
+        Warning at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:350): unexpected error has occurred, but yours did!
+        Error at edu.wpi.first.wpilibj.RobotBase.runRobot(RobotBase.java:352): The startCompetition() method (or methods called by it) should have handled the exception above.
+
+      Reading the stack trace, you can see that the issue happened inside of the ``robotInit()`` function, on line 24, and the exception involved "Null Pointer".
+
+      By going to line 24, you can see there is only one thing which could be null - ``armMotorCtrl``. Looking further up, you can see that the ``armMotorCtrl`` object is declared, but never instantiated.
+
+      Alternatively, you can step through lines of code with the single step debugger, and stop when you hit line 24. Inspecting the ``armMotorCtrl`` object at that point would show that it is null.
+
+   .. group-tab:: C++
+
+      .. code-block:: text
+
+         Exception has occurred: W32/0xc0000005
```

Comment:
Might be worth mentioning that this error will look different on the RIO / other non-Windows platforms.